pkg/storage/imagestore/imagestore.go

```diff
@@ -1689,40 +1692,54 @@ func (is *ImageStore) deleteBlob(repo string, digest godigest.Digest) error {
 	return nil
 }
 
-func getBlobDigest(imgStore *ImageStore, path string) (godigest.Digest, error) {
+func getBlobDigest(imgStore *ImageStore, path string, digestAlgorithm godigest.Algorithm,
+) (godigest.Digest, error) {
 	fileReader, err := imgStore.storeDriver.Reader(path, 0)
 	if err != nil {
 		return "", zerr.ErrUploadNotFound
 	}
 
 	defer fileReader.Close()
 
-	digest, err := godigest.FromReader(fileReader)
+	digest, err := digestAlgorithm.FromReader(fileReader)
 	if err != nil {
 		return "", zerr.ErrBadBlobDigest
 	}
 
 	return digest, nil
 }
 
-func (is *ImageStore) GetAllBlobs(repo string) ([]string, error) {
-	dir := path.Join(is.rootDir, repo, "blobs", "sha256")
+func (is *ImageStore) GetAllBlobs(repo string) ([]godigest.Digest, error) {
+	blobsDir := path.Join(is.rootDir, repo, ispec.ImageBlobsDir)
 
-	files, err := is.storeDriver.List(dir)
-	if err != nil {
-		if errors.As(err, &driver.PathNotFoundError{}) {
-			is.log.Debug().Msg("empty rootDir")
+	algorithms := []godigest.Algorithm{
+		godigest.SHA256,
```

Comment:
Do you really want to list the approved algorithms or let godigest do that for you.  When you create a `godigest.Algorithm` you can call `alg.Available()` to see if the digest algorithm has been registered.  I do not believe the spec limits the algorithms allowed (it only requires sha256).   Then other algorithms can be added without modifying this code.   godigest allows other algorithms to be registered with [godigest.RegisterAlgorithm](https://github.com/opencontainers/go-digest/blob/master/algorithm.go#L78).  There is a PR to add blake3 to the list of algorithms and that should ideally not require any changes to the Zot code based except a version bump on godigest.

Reply:
The issue is the godigest library doesn't have an API to retrieve a slice of all registered algorithms. So I ended up listing them by hand. 

I also made the assumption these algorithms here are always available, as they are defined as constants there in the library. Is this correct?

Reply:
Worked around this issue by checking the names of the existing folders and checking if they are available, as opposed to obtaining the list of available algorithms first.